sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracer.java

```diff
@@ -19,7 +19,7 @@ final class SdkTracer implements ExtendedTracer {
 
   private final TracerSharedState sharedState;
   private final InstrumentationScopeInfo instrumentationScopeInfo;
-  private final boolean tracerEnabled;
+  private boolean tracerEnabled;
```

Comment:
Maybe something like this would do the trick @jackshirazi 
```suggestion
  // TODO: add dedicated API for updating scope config.
  @SuppressWarnings("FieldCanBeFinal") // For now, allow updating reflectively.
  private boolean tracerEnabled;
```